segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/noop/ConditionalNoOpChunkStorage.java

```diff
@@ -0,0 +1,155 @@
+        this.inner = inner;
+        this.noOpChunkStorage = new NoOpChunkStorage(executorService);
+        this.executorService = executorService;
+    }
+
+    @Override
+    public boolean supportsTruncation() {
+        return inner.supportsTruncation();
+    }
+
+    @Override
+    public boolean supportsAppend() {
+        return inner.supportsAppend();
+    }
+
+    @Override
+    public boolean supportsConcat() {
+        return inner.supportsConcat();
+    }
+
+    @Override
+    public CompletableFuture<Boolean> exists(String chunkName) {
+        return isMetadataCall(getChunkHandle(chunkName)) ? inner.exists(chunkName) : noOpChunkStorage.exists(chunkName);
+    }
+
+    /**
+     * Creates a chunk in the chunkStorage.
+     *
+     * @param chunkName String name of the storage object to create.
+     * @return handle to a chunk
+     */
+    @Override
+    public CompletableFuture<ChunkHandle> create(String chunkName) {
+        return isMetadataCall(getChunkHandle(chunkName)) ? inner.create(chunkName) : noOpChunkStorage.create(chunkName);
+    }
+
+    @Override
+    public CompletableFuture<ChunkHandle> createWithContent(String chunkName, int length, InputStream data) {
+        return isMetadataCall(getChunkHandle(chunkName)) ? inner.createWithContent(chunkName, length, data) : noOpChunkStorage.createWithContent(chunkName, length, data);
+    }
+
+    @Override
+    public CompletableFuture<Void> delete(ChunkHandle handle) {
+        return isMetadataCall(handle) ? inner.delete(handle) : noOpChunkStorage.delete(handle);
+    }
+
+    @Override
+    public CompletableFuture<ChunkHandle> openRead(String chunkName) {
+        return isMetadataCall(getChunkHandle(chunkName)) ? inner.openRead(chunkName) : noOpChunkStorage.openRead(chunkName);
+    }
+
+    @Override
+    public CompletableFuture<ChunkHandle> openWrite(String chunkName) {
+        return isMetadataCall(getChunkHandle(chunkName)) ? inner.openWrite(chunkName) : noOpChunkStorage.openWrite(chunkName);
+    }
+
+    @Override
+    public CompletableFuture<ChunkInfo> getInfo(String chunkName) {
+        return isMetadataCall(getChunkHandle(chunkName)) ? inner.getInfo(chunkName) : noOpChunkStorage.getInfo(chunkName);
+    }
+
+    @Override
+    public CompletableFuture<Integer> read(ChunkHandle handle, long fromOffset, int length, byte[] buffer, int bufferOffset) {
+        return isMetadataCall(handle) ? inner.read(handle, fromOffset, length, buffer, bufferOffset) : noOpChunkStorage.read(handle, fromOffset, length, buffer, bufferOffset);
+    }
+
+    @Override
+    public CompletableFuture<Integer> write(ChunkHandle handle, long offset, int length, InputStream data) {
+        return isMetadataCall(handle) ? inner.write(handle, offset, length, data) : noOpChunkStorage.write(handle, offset, length, data);
+    }
+
+    @Override
+    public CompletableFuture<Integer> concat(ConcatArgument[] chunks) {
+        return inner.concat(chunks);
```

Comment:
This is missing isMetadataCall check
should be.
`return isMetadataCall(handle) ? inner.concat(chunks) : noOpChunkStorage.concat(chunks);`

Also not sure why this method is not covered

Reply:
Done